src/common/models/config.model.ts

```diff
@@ -177,20 +177,42 @@ export interface IConfigCustomAction extends IConfigMethod {
   fields: IConfigInputField[]
 }
 
-export interface IConfigPagination {
-  source: 'query'
+export type IConfigPagination = 
+  IConfigQueryPagination | 
+  IConfigJSONBodyPagination;
+
+export type IConfigQueryPagination = _IConfigPagination<'query', IConfigQueryPaginationParams>;
+export type IConfigJSONBodyPagination = _IConfigPagination<'json-body', IConfigJSONBodyPaginationParams>;
+
+export const isQueryPagination = (obj: IConfigPagination): obj is IConfigQueryPagination => {
```

Comment:
Please move these functions to another file, as the `config.model` file should only describe the model.

Reply:
Where do you propose to move these? These do the type casting for the interfaces.

Reply:
Hmm how about exporting all of this type checking methods to an external helper file?

Reply:
I would be fine with that as well. `config.helpers.type.ts` ?

Reply:
`config.types.helper.ts`?